messaging/src/test/java/org/axonframework/tracing/NestingSpanFactoryTest.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2010-2022. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.tracing;
+
+import org.axonframework.common.AxonConfigurationException;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.eventhandling.GenericEventMessage;
+import org.axonframework.messaging.Message;
+import org.junit.jupiter.api.*;
+import org.mockito.*;
+
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.function.Supplier;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+class NestingSpanFactoryTest {
+
+    private final Clock clock = Mockito.spy(Clock.class);
+    private final SpanFactory delegate = Mockito.mock(SpanFactory.class);
+    private final Span mockSpan = Mockito.mock(Span.class);
+    private final NestingSpanFactory spanFactory = NestingSpanFactory.builder()
+                                                                     .clock(clock)
+                                                                     .delegate(delegate)
+                                                                     .build();
+    private final Supplier<String> nameSupplier = () -> "spanName";
+    EventMessage<Object> message = GenericEventMessage.asEventMessage("payload");
```

Comment:
Why is the `message` package-private?
```suggestion
    private final EventMessage<Object> message = GenericEventMessage.asEventMessage("payload");
```